x-pack/plugins/synthetics/public/apps/synthetics/hooks/use_breadcrumbs.ts

```diff
@@ -55,28 +57,35 @@ export const makeBaseBreadcrumb = (
     uptimePath += query === EMPTY_QUERY ? '' : query;
   }
 
-  return [
-    {
+  const baseBreadcrumbs: EuiBreadcrumb[] = [];
+
+  // serverless Kibana has a curated UX flow, and "Observability" is already a given,
+  // thus we don't need to include it explicitly in the breadcrumb trail
+  if (!isServerless) {
+    baseBreadcrumbs.push({
       text: i18n.translate('xpack.synthetics.breadcrumbs.observabilityText', {
         defaultMessage: 'Observability',
       }),
       href: observabilityPath,
       'data-test-subj': 'observabilityPathBreadcrumb',
-    },
-    {
-      text: i18n.translate('xpack.synthetics.breadcrumbs.overviewBreadcrumbText', {
-        defaultMessage: 'Synthetics',
-      }),
-      href: uptimePath,
-      'data-test-subj': 'syntheticsPathBreadcrumb',
-    },
-  ];
+    });
+  }
+
+  baseBreadcrumbs.push({
+    text: i18n.translate('xpack.synthetics.breadcrumbs.overviewBreadcrumbText', {
+      defaultMessage: 'Synthetics',
+    }),
+    href: uptimePath,
+    'data-test-subj': 'syntheticsPathBreadcrumb',
+  });
+
+  return baseBreadcrumbs;
 };
 
 export const useBreadcrumbs = (extraCrumbs: ChromeBreadcrumb[]) => {
   const params = useUrlParams()[0]();
   const kibana = useKibana();
-  const setBreadcrumbs = kibana.services.chrome?.setBreadcrumbs;
+  const { setBreadcrumbs, isServerless } = useContext(SyntheticsSettingsContext);
```

Comment:
@crespocarlos I liked what @justinkambic is doing here. He's abstracting the `setBreadcrumbs` during kibana start to be either serverless or stateful kibana.

Clients don't care where they are, they just know how to add its value to the breadcrumbs.

I see the exact same code being written in different ways all over our plugins. Maybe this is a good thing to export to a common place.

Reply:
Yeah, this is very nice and more interesting than the other approaches I saw in the codebase. 
At first, I thought about following a similar path, but I didn't want to change the `useBreadcrumbs` in the `observabilty_shared` at this moment. 
Especially because by doing that, we're overriding the default serverless breadcrumbs feature, which observes the navigation tree config in the `serverless-*` plugins and sets the breadcrumbs automatically. If we do the same in the `observability_shared` we need to make sure everything that depends on it still works as expected